payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt

```diff
@@ -1,32 +1,84 @@
 package com.stripe.android.payments.paymentlauncher
 
+import android.content.Context
+import androidx.activity.result.ActivityResultCaller
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
+import com.stripe.android.StripeIntentResult
+import com.stripe.android.exception.APIException
 import com.stripe.android.model.ConfirmStripeIntentParams
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.view.AuthActivityStarterHost
 
 /**
  * WIP - view model for PaymentLauncherHostActivity
  */
+
 internal class PaymentLauncherViewModel : ViewModel() {
+    /**
+     * [PaymentResult] live data to be observed.
+     */
+    internal val paymentLauncherResult = MutableLiveData<PaymentResult>()
+
+    /**
+     * Register for [ActivityResultCaller] when it's available.
+     */
+    internal fun registerFromActivity(activityResultCaller: ActivityResultCaller) {}
+
+    /**
+     * Unregister any [ActivityResultCaller] hosted.
+     */
+    internal fun unregisterFromActivity() {}
```

Comment:
Why do we need these?

Reply:
oh i'm trying to replicate what we did in `PaymentSheetViewModel`, where the it outlives across Activity, but just realized here we can just create a new one in each new `PaymentLauncherConfirmationActivity` here, updated and removed